From 22f5c18762194cef83adf486705099f8a1985266 Mon Sep 17 00:00:00 2001
From: Peter Varga <pvarga@inf.u-szeged.hu>
Date: Wed, 14 May 2025 13:08:16 +0200
Subject: [PATCH 07/13] UPSTREAM: Remove --enable-webgl-software-rendering

Most probably it is useless with the new default ANGLE backend.
The original code has been trimmed and now it is just equivalent with
passing the following command line arguments:
--webEngineArgs --disable-gpu-rasterization --ignore-gpu-blocklist

As a drive-by, change WebEngineContext::initCommandLine() parameter to
pointer instead of reference to make it clear on the caller side it will
not use but it is expected to change the parameter's value.

Pick-to: 6.10
Change-Id: Ic118fa55071c5d8ac0730db2f640c3132375db8b
Reviewed-by: Allan Sandfeld Jensen <allan.jensen@qt.io>
---
 qtwebengine/src/core/web_engine_context.cpp | 22 +++++----------------
 qtwebengine/src/core/web_engine_context.h   |  3 +--
 2 files changed, 6 insertions(+), 19 deletions(-)

diff --git a/qtwebengine/src/core/web_engine_context.cpp b/qtwebengine/src/core/web_engine_context.cpp
index a6e050d0d7..92774bde98 100644
--- a/qtwebengine/src/core/web_engine_context.cpp
+++ b/qtwebengine/src/core/web_engine_context.cpp
@@ -813,10 +813,8 @@ WebEngineContext::WebEngineContext()
     // Allow us to inject javascript like any webview toolkit.
     content::RenderFrameHost::AllowInjectingJavaScript();
 
-    bool useEmbeddedSwitches = false;
-    bool enableGLSoftwareRendering = false;
-    base::CommandLine *parsedCommandLine =
-            initCommandLine(useEmbeddedSwitches, enableGLSoftwareRendering);
+    bool useEmbeddedSwitches;
+    base::CommandLine *parsedCommandLine = initCommandLine(&useEmbeddedSwitches);
 
     setupProxyPac(parsedCommandLine);
     parsedCommandLine->AppendSwitchPath(switches::kBrowserSubprocessPath, WebEngineLibraryInfo::getPath(content::CHILD_PROCESS_EXE));
@@ -946,10 +944,6 @@ WebEngineContext::WebEngineContext()
 
     initializeFeatureList(parsedCommandLine, enableFeatures, disableFeatures);
 
-    // If user requested GL support instead of using Skia rendering to
-    // bitmaps, use software rendering via software OpenGL. This might be less
-    // performant, but at least provides WebGL support.
-    // TODO(miklocek), check if this still works with latest chromium
     const bool disableGpu = parsedCommandLine->HasSwitch(switches::kDisableGpu);
     std::string glType;
     if (parsedCommandLine->HasSwitch(switches::kUseGL))
@@ -962,10 +956,6 @@ WebEngineContext::WebEngineContext()
     parsedCommandLine->AppendSwitch(switches::kInProcessGPU);
 
     if (glType != gl::kGLImplementationDisabledName) {
-        if (enableGLSoftwareRendering) {
-            parsedCommandLine->AppendSwitch(switches::kDisableGpuRasterization);
-            parsedCommandLine->AppendSwitch(switches::kIgnoreGpuBlocklist);
-        }
         if (glType != gl::kGLImplementationANGLEName) {
             qWarning("Only --use-gl=angle is supported on this platform.");
         }
@@ -1066,8 +1056,7 @@ printing::PrintJobManager* WebEngineContext::getPrintJobManager()
 }
 #endif
 
-base::CommandLine *WebEngineContext::initCommandLine(bool &useEmbeddedSwitches,
-                                                     bool &enableGLSoftwareRendering)
+base::CommandLine *WebEngineContext::initCommandLine(bool *useEmbeddedSwitches)
 {
     if (!base::CommandLine::CreateEmpty())
         qFatal("base::CommandLine has been initialized unexpectedly.");
@@ -1094,11 +1083,10 @@ base::CommandLine *WebEngineContext::initCommandLine(bool &useEmbeddedSwitches,
         }
     }
 #if defined(QTWEBENGINE_EMBEDDED_SWITCHES)
-    useEmbeddedSwitches = !appArgs.contains("--disable-embedded-switches"_L1);
+    *useEmbeddedSwitches = !appArgs.contains("--disable-embedded-switches"_L1);
 #else
-    useEmbeddedSwitches = appArgs.contains("--enable-embedded-switches"_L1);
+    *useEmbeddedSwitches = appArgs.contains("--enable-embedded-switches"_L1);
 #endif
-    enableGLSoftwareRendering = appArgs.removeAll("--enable-webgl-software-rendering"_L1);
     appArgs.removeAll("--disable-embedded-switches"_L1);
     appArgs.removeAll("--enable-embedded-switches"_L1);
 
diff --git a/qtwebengine/src/core/web_engine_context.h b/qtwebengine/src/core/web_engine_context.h
index 6f2bf04c4b..f95f582487 100644
--- a/qtwebengine/src/core/web_engine_context.h
+++ b/qtwebengine/src/core/web_engine_context.h
@@ -80,8 +80,7 @@ public:
     void removeProfileAdapter(ProfileAdapter *profileAdapter);
     bool profileExistOnPath(const QString &dataPath);
     void destroy();
-    static base::CommandLine *initCommandLine(bool &useEmbeddedSwitches,
-                                              bool &enableGLSoftwareRendering);
+    static base::CommandLine *initCommandLine(bool *useEmbeddedSwitches);
 
 private:
     friend class base::RefCounted<WebEngineContext>;
-- 
2.51.0

